# Overview
This is the GitHub repository for the following ongoing project:

**Investigating the Dynamics of RSV Genetic Diversity during an Epidemic Cycle**

# Abstract

## Background and aims
The genome of Respiratory Syncytial Virus (RSV) contains 10 genes encoding 11 proteins. RSV is divided into two antigenic subgroups, A and B, primarily based on differences in the attachment (G) glycoprotein. Both subtypes co-circulate, but one typically dominates during an epidemic season. While the genetic variability and evolutionary patterns of RSV at the genotype level have been studied, gaps remain in understanding the diversity of circulating lineages within a single season, and the mechanisms behind shifts in subgroup dominance between seasons. This study evaluates genetic diversity and population dynamics across multiple seasons to determine the spread of RSV-A and -B lineages within an epidemic cycle, with a focus on sequences from Germany.

## Methods
We analyzed 168 RSV-A and 135 RSV-B whole genome sequences (WGS) from Germany, alongside 563 RSV-A and 564 RSV-B sequences from other European countries retrieved from NCBI GenBank. Only sequences with collection dates from the 2014/15 season to the 2022/23 season were considered. Non-phylogenetic methods assessed pairwise genetic diversity using single nucleotide polymorphism (SNP) and evolutionary distances with an 8-week sliding window. Bayesian phylogenetic analysis was conducted, and time-stamped phylogenetic trees and Bayesian SkyGrids were used to assess temporal and population dynamics of RSV-A and -B. 

## Results 
For German RSV sequences, we observed a decrease in average pairwise distance of the dominant subtype (e.g. RSV-A) towards the end of the season if a shift in type dominance occurred in the following season (e.g. RSV-A to RSV-B). This pattern was also observed in sequences from Spain. RSV-B generally showed lower variation in genetic diversity than RSV-A. Phylogenetic analysis indicated that German RSV-A and RSV-B sequences from a single season typically belong to multiple clades. However, we observed that German post-COVID-19 pandemic RSV-A and RSV-B sequences each derived from a single monophyletic clade. 

## Conclusion
Examining the temporal pattern of genetic diversity within an epidemic cycle could demonstrate the circulation of RSV lineages throughout a season and reveal potential indicators for subtype dominance changes before the following season. The increased resolution achieved through this analysis provides further insight for future RSV surveillance and public health strategies. 

## Authors
Ina Li, RWTH Aachen University, Germany / Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Jiye Kwon, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Verity Hill, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Ke Li, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Yi Ting Chew, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Virginia E. Pitzer, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Daniel M. Weinberger, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.