# Overview
This is the GitHub repository for the following ongoing project:

**Investigating the Dynamics of RSV Genetic Diversity during an Epidemic Cycle**

# Abstract

## Background and aims
Respiratory syncytial virus (RSV) genome contains 10 genes encoding 11 proteins. RSV is divided into two antigenic subgroups, A and B, primarily based on differences in the attachment (G) glycoprotein. While both subtypes co-circulate, one typically dominates during an epidemic season. The genetic variability and evolutionary patterns of RSV at the genotype level have been studied. However, gaps remain in understanding the diversity of circulating lineages within a single season and the mechanisms behind shifts in subgroup dominance between seasons. We evaluate genetic diversity and population dynamics across multiple seasons to determine the spread of RSV-A and -B lineages within an epidemic cycle, with a focus on German sequences within European context.

## Methods
We analyzed 168 RSV-A and 135 RSV-B whole genome sequences (WGS) from Germany, alongside 563 RSV-A and 564 RSV-B sequences from other European countries retrieved from NCBI GenBank. Only sequences with specified collection dates from season 2014/15 to the 2022/23 season were considered. Non-phylogenetic methods assessed pairwise genetic diversity using single nucleotide polymorphism (SNP) and evolutionary distances with an 8-week sliding window. Phylogenetic analysis was conducted using Bayesian Evolutionary Analysis Sampling Trees (BEAST) software to build time-stamped phylogenetic trees and Bayesian SkyGrids was used to assess temporal and population dynamics of RSV-A and -B.  

## Results 
Our analysis reveals a visible decrease in the average pairwise distance of the dominant subtype (e.g. RSV-A) towards the end of the season when a shift in dominance is expected in the next season (e.g. from RSV-A to RSV-B). RSV-B generally shows lower variation in genetic diversity than RSV-A. Both evolutionary and SNP distances exhibited consistent patterns across German and other European sequences. Additional phylogenetic results are forthcoming. 

## Conclusion
Examining the temporal pattern of genetic diversity within an epidemic cycle could demonstrate the circulation of RSV lineages throughout a season and reveals potential indicators for subtype dominance changes before the following season. The increased resolution achieved through this analysis provides further insight for future RSV surveillance and public health strategies.

## Authors
Ina Li, RWTH Aachen University, Germany / Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Jiye Kwon, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Verity Hill, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Ke Li, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Yi Ting Chew, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Virginia E. Pitzer, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Daniel Weinberger, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.