# Overview
This is the GitHub respository for the following ongoing project:

**Investigating the Dynamics of RSV Genetic Diversity during an Epidemic Cycle**

# Abstract

## Background and aims
Respiratory syncytial virus (RSV) is divided into two antigenically distinct subtypes, 
RSV-A and RSV-B, based on differences in the attachment glycoprotein (G). While both subtypes co-circulate, one typically dominates during an epidemic season. The genetic diversity of RSV is observed with genotyping based on the G protein, but the change in diversity of circulating lineages within a single season and the mechanics of subtype dominance shift between seasons remain unexplored. By using whole genome sequences (WGS) to determine genetic diversity dynamics, phylogeny and population dynamics across multiple seasons, we investigate the transmission patterns of RSV-A and -B lineages within an epidemic cycle, with a focus on German sequences within European context.  

## Methods
We analyzed 168 RSV-A and 135 RSV-B sequences from Germany, alongside 563 RSV-A and 564 RSV-B sequences from other European countries retrieved from NCBI GenBank. WGS (> 90% completeness) with collection dates from season 2014/15 to most recent available data from 2022/23 were considered. Pairwise genetic diversity was measured using SNP distance and evolutionary distance for nucleotide sequences, while Hamming distance was calculated for the corresponding protein sequences. To discern the temporal dynamics of genetic diversity, an 8-week sliding window approach was implemented for both subtypes and each diversity measure. The software package BEAST was used for constructing time-stamped phylogenetic trees and Bayesian SkyGrids to determine phylogeny and population dynamics. 

## Results 
The observed trend in RSV genetic diversity of German sequences indicates that the average pairwise distance of the dominant subtype decreases towards the end of the season if a dominance switch is imminent for the following season. Compared to RSV-A, RSV-B generally shows lower variation in genetic diversity. Both evolutionary and SNP distances exhibited the same trends for German and other European sequences. Phylogenetic results will be presented later as the study continues. 

## Conclusion
Examining the temporal pattern of genetic diversity within an epidemic cycle demonstrates the transmission of RSV lineages throughout a season and reveals potential indicators for subtype dominance changes before the following season. The increased resolution achieved through this analysis provides further insight for future RSV surveillance and public health strategies.

## Authors
Ina Li, RWTH Aachen University, Germany.

Jiye Kwon, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Verity Hill, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Ke Li, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Yi Ting Chew, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Daniel Weinberger, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.

Virginia E. Pitzer, Department of Epidemiology of Microbial Diseases, Yale School of Public Health, USA.
